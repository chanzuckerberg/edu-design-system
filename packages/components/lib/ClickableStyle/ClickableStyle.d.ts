import React from "react";
export declare type ClickableStyleProps<IComponent extends React.ElementType> = {
    /**
     * CSS class for custom styles.
     */
    className: string;
    /**
     * The color of the element.
     */
    color: "alert" | "brand" | "neutral" | "success" | "warning";
    /**
     * The size of the element.
     */
    size: "small" | "medium" | "large";
    /**
     * A hidden prop for visual testing
     */
    state?: "inactive" | "hover" | "focus" | "active";
    /**
     * The style of the element.
     */
    variant: "flat" | "outline" | "link" | "plain";
} & React.ComponentProps<IComponent>;
/**
 * A helper component that contains all the styling for buttons and links.
 *
 * If you're styling a `<button>` or `<a>` element, you can use the `Button`
 * and `Link` components (respectively). `ClickableStyle` should only be used
 * directly when styling other elements or components (e.g. `Link` from `react-router`)
 * to look like a button or link.
 *
 * See the `Button` and `Link` stories for usage examples.
 */
declare const ClickableStyle: React.ForwardRefExoticComponent<Pick<any, string | number | symbol> & React.RefAttributes<HTMLElement>>;
export default ClickableStyle;
