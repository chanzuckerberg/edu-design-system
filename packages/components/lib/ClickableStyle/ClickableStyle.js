"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const clsx_1 = __importDefault(require("clsx"));
const react_1 = __importDefault(require("react"));
const ClickableStyle_module_css_1 = __importDefault(require("./ClickableStyle.module.css"));
/**
 * A helper component that contains all the styling for buttons and links.
 *
 * If you're styling a `<button>` or `<a>` element, you can use the `Button`
 * and `Link` components (respectively). `ClickableStyle` should only be used
 * directly when styling other elements or components (e.g. `Link` from `react-router`)
 * to look like a button or link.
 *
 * See the `Button` and `Link` stories for usage examples.
 */
const ClickableStyle = react_1.default.forwardRef((_a, ref) => {
    var { as: Component, children, color, size, state, variant, className } = _a, rest = __rest(_a, ["as", "children", "color", "size", "state", "variant", "className"]);
    return (react_1.default.createElement(Component, Object.assign({ className: (0, clsx_1.default)(className, ClickableStyle_module_css_1.default.button, 
        // Sizes
        variant !== "link" && [
            size === "small" && ClickableStyle_module_css_1.default.sizeSmall,
            size === "medium" && ClickableStyle_module_css_1.default.sizeMedium,
            size === "large" && ClickableStyle_module_css_1.default.sizeLarge,
        ], 
        // Variants
        variant === "flat" && ClickableStyle_module_css_1.default.variantFlat, variant === "outline" && ClickableStyle_module_css_1.default.variantOutline, variant === "link" && ClickableStyle_module_css_1.default.variantLink, variant === "plain" && ClickableStyle_module_css_1.default.variantPlain, 
        // Colors
        color === "alert" && ClickableStyle_module_css_1.default.colorAlert, color === "brand" && ClickableStyle_module_css_1.default.colorBrand, color === "neutral" && ClickableStyle_module_css_1.default.colorNeutral, color === "success" && ClickableStyle_module_css_1.default.colorSuccess, color === "warning" && ClickableStyle_module_css_1.default.colorWarning, 
        // Interactive States (for testing)
        state === "hover" && ClickableStyle_module_css_1.default.stateHover, state === "focus" && ClickableStyle_module_css_1.default.stateFocus, state === "active" && ClickableStyle_module_css_1.default.stateActive), ref: ref }, rest), children));
});
ClickableStyle.displayName = "ClickableStyle";
exports.default = ClickableStyle;
//# sourceMappingURL=ClickableStyle.js.map