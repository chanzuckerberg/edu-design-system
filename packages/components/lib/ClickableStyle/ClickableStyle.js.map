{"version":3,"file":"ClickableStyle.js","sourceRoot":"","sources":["../../src/ClickableStyle/ClickableStyle.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;AAAA,gDAAwB;AACxB,kDAA0B;AAC1B,4FAAiD;AAyBjD;;;;;;;;;GASG;AACH,MAAM,cAAc,GAAG,eAAK,CAAC,UAAU,CACrC,CACE,EASkC,EAClC,GAAoC,EACpC,EAAE;QAXF,EACE,EAAE,EAAE,SAAS,EACb,QAAQ,EACR,KAAK,EACL,IAAI,EACJ,KAAK,EACL,OAAO,EACP,SAAS,OAEuB,EAD7B,IAAI,cART,oEASC,CADQ;IAIT,OAAO,CACL,8BAAC,SAAS,kBACR,SAAS,EAAE,IAAA,cAAI,EACb,SAAS,EACT,mCAAM,CAAC,MAAM;QACb,QAAQ;QACR,OAAO,KAAK,MAAM,IAAI;YACpB,IAAI,KAAK,OAAO,IAAI,mCAAM,CAAC,SAAS;YACpC,IAAI,KAAK,QAAQ,IAAI,mCAAM,CAAC,UAAU;YACtC,IAAI,KAAK,OAAO,IAAI,mCAAM,CAAC,SAAS;SACrC;QACD,WAAW;QACX,OAAO,KAAK,MAAM,IAAI,mCAAM,CAAC,WAAW,EACxC,OAAO,KAAK,SAAS,IAAI,mCAAM,CAAC,cAAc,EAC9C,OAAO,KAAK,MAAM,IAAI,mCAAM,CAAC,WAAW,EACxC,OAAO,KAAK,OAAO,IAAI,mCAAM,CAAC,YAAY;QAC1C,SAAS;QACT,KAAK,KAAK,OAAO,IAAI,mCAAM,CAAC,UAAU,EACtC,KAAK,KAAK,OAAO,IAAI,mCAAM,CAAC,UAAU,EACtC,KAAK,KAAK,SAAS,IAAI,mCAAM,CAAC,YAAY,EAC1C,KAAK,KAAK,SAAS,IAAI,mCAAM,CAAC,YAAY,EAC1C,KAAK,KAAK,SAAS,IAAI,mCAAM,CAAC,YAAY;QAC1C,mCAAmC;QACnC,KAAK,KAAK,OAAO,IAAI,mCAAM,CAAC,UAAU,EACtC,KAAK,KAAK,OAAO,IAAI,mCAAM,CAAC,UAAU,EACtC,KAAK,KAAK,QAAQ,IAAI,mCAAM,CAAC,WAAW,CACzC,EACD,GAAG,EAAE,GAAG,IACJ,IAAI,GAEP,QAAQ,CACC,CACb,CAAC;AACJ,CAAC,CACF,CAAC;AAEF,cAAc,CAAC,WAAW,GAAG,gBAAgB,CAAC;AAE9C,kBAAe,cAAc,CAAC"}