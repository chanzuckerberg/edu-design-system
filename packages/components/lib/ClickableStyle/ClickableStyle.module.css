.button {
  /* Base Styles */

  /* Using motion-reduce:transition-none here because motion-safe:transition-colors doesn't seem to
     be having an effect. :( */

  display: inline-flex;

  align-items: center;

  border-radius: 0.25rem;

  border-width: 2px;

  transition-property: background-color, border-color, color, fill, stroke;

  transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);

  transition-duration: 150ms;
}@media (prefers-reduced-motion: reduce) {.button {

    transition-property: none;
  }
}.button {

  /* Base Text Styles */

  font-weight: 700;

  /* Default variable values */
  --plain-background-color: transparent;
}

/* Style icons that may be in the button children */

.button svg {
  --svg-icon-size: 2em;
}

/* Component Sizes */

.sizeSmall {

  height: 1.5rem;

  padding-left: 0.25rem;

  padding-right: 0.25rem;

  font-size: 0.75rem;

  line-height: 1rem;

  line-height: .75rem;
}

.sizeMedium {

  height: 2rem;

  padding-left: 1rem;

  padding-right: 1rem;

  font-size: 0.75rem;

  line-height: 1rem;

  line-height: .75rem;
}

.sizeLarge {

  height: 2.5rem;

  padding-left: 1rem;

  padding-right: 1rem;

  font-size: 0.875rem;

  line-height: 1.25rem;
}

/* Button Colors */

.colorBrand {
  --primary-color: var(--eds-color-brand-600);
  --link-color: var(--eds-color-brand-700);
}

.colorBrand:hover,
  .colorBrand.stateHover,
  .colorBrand:focus,
  .colorBrand.stateFocus {
    --primary-color: var(--eds-color-brand-700);
    --link-color: var(--eds-color-brand-800);
    --plain-background-color: var(--eds-color-brand-100);
  }

.colorBrand:active,
  .colorBrand.stateActive {
    --primary-color: var(--eds-color-brand-800);
    --plain-background-color: var(--eds-color-brand-600);
  }

.colorAlert {
  --primary-color: var(--eds-color-alert-600);
  --link-color: var(--eds-color-alert-700);
}

.colorAlert:hover,
  .colorAlert.stateHover,
  .colorAlert:focus,
  .colorAlert.stateFocus {
    --primary-color: var(--eds-color-alert-700);
    --link-color: var(--eds-color-alert-800);
    --plain-background-color: var(--eds-color-alert-100);
  }

.colorAlert:active,
  .colorAlert.stateActive {
    --primary-color: var(--eds-color-alert-800);
    --plain-background-color: var(--eds-color-alert-600);
  }

.colorNeutral {
  --primary-color: var(--eds-color-neutral-500);
  --link-color: var(--eds-color-neutral-600);
}

.colorNeutral:hover,
  .colorNeutral.stateHover,
  .colorNeutral:focus,
  .colorNeutral.stateFocus {
    --primary-color: var(--eds-color-neutral-600);
    --link-color: var(--eds-color-neutral-700);
    --plain-background-color: var(--eds-color-neutral-100);
  }

.colorNeutral:active,
  .colorNeutral.stateActive {
    --primary-color: var(--eds-color-neutral-700);
    --plain-background-color: var(--eds-color-neutral-500);
  }

.colorSuccess {
  --primary-color: var(--eds-color-success-600);
  --link-color: var(--eds-color-success-700);
}

.colorSuccess:hover,
  .colorSuccess.stateHover,
  .colorSuccess:focus,
  .colorSuccess.stateFocus {
    --primary-color: var(--eds-color-success-700);
    --link-color: var(--eds-color-success-800);
    --plain-background-color: var(--eds-color-success-100);
  }

.colorSuccess:active,
  .colorSuccess.stateActive {
    --primary-color: var(--eds-color-success-800);
    --plain-background-color: var(--eds-color-success-600);
  }

.colorWarning {
  --primary-color: var(--eds-color-warning-600);
  --link-color: var(--eds-color-warning-700);
}

.colorWarning:hover,
  .colorWarning.stateHover,
  .colorWarning:focus,
  .colorWarning.stateFocus {
    --primary-color: var(--eds-color-warning-700);
    --link-color: var(--eds-color-warning-800);
    --plain-background-color: var(--eds-color-warning-100);
  }

.colorWarning:active,
  .colorWarning.stateActive {
    --primary-color: var(--eds-color-warning-800);
    --plain-background-color: var(--eds-color-warning-600);
  }

.variantFlat {
  background-color: var(--primary-color);
  border-color: var(--primary-color);
  color: var(--eds-color-white);
}

.variantOutline {

  background-color: transparent;

  border-color: var(--link-color);
  color: var(--link-color);
}

.variantOutline:hover, .variantOutline.stateHover, .variantOutline:focus, .variantOutline.stateFocus, .variantOutline:active, .variantOutline.stateActive {

  color: var(--eds-color-white);
}

.variantOutline:hover,
  .variantOutline.stateHover,
  .variantOutline:focus,
  .variantOutline.stateFocus,
  .variantOutline:active,
  .variantOutline.stateActive {

    background-color: var(--primary-color);
    border-color: var(--primary-color);
  }

/* override the hover/focus/active values */

.variantOutline:disabled {

  background-color: transparent;
}

.variantOutline:disabled {

    border-color: var(--primary-color);
    color: var(--primary-color);
  }

.variantLink {

  border-style: none;

  padding-left: 0px;

  padding-right: 0px;

  padding-top: 0px;

  padding-bottom: 0px;

  font-weight: 700;

  text-decoration: underline;

  color: var(--link-color);

  /* button should defer to surrounding text for size/line-height/case */
  font-size: inherit;
  line-height: inherit;
  text-transform: inherit;
}

.variantLink:hover,
  .variantLink.stateHover,
  .variantLink:focus,
  .variantLink.stateFocus {
    text-decoration-thickness: 2px;
  }

.variantLink:active,
  .variantLink.stateActive,
  .variantLink:disabled {
    color: var(--primary-color);
  }

/* override the hover/focus values */

.variantLink:disabled {
    text-decoration-thickness: auto;
  }

.variantPlain {
  color: var(--link-color);
  border-color: var(--plain-background-color);
  background-color: var(--plain-background-color);
}

.variantPlain:hover,
  .variantPlain.stateHover,
  .variantPlain:focus,
  .variantPlain.stateFocus {
    color: var(--primary-color);
    border-color: var(--primary-color);
  }

.variantPlain:active, .variantPlain.stateActive {

  color: var(--eds-color-white);
}

.variantPlain:active,
  .variantPlain.stateActive {

    border-color: var(--plain-background-color);
  }

.variantPlain:disabled {

  border-color: transparent;
}

.variantPlain:disabled {

    color: var(--primary-color);
  }

/* Link icons should be the same size as the text */

.variantLink svg {
  --svg-icon-size: 1em;
}

/* comes after other styles to override hover behaviors */

.button:focus,
.button.stateFocus {
  /* Focus Outline */
  outline: 2px solid transparent;
  outline-offset: 2px;
  --tw-ring-offset-shadow: var(--tw-ring-inset) 0 0 0 var(--tw-ring-offset-width) var(--tw-ring-offset-color);
  --tw-ring-shadow: var(--tw-ring-inset) 0 0 0 calc(3px + var(--tw-ring-offset-width)) var(--tw-ring-color);
  box-shadow: var(--tw-ring-offset-shadow), var(--tw-ring-shadow), var(--tw-shadow, 0 0 #0000);
  --tw-ring-color: var(--eds-color-info-400);
}

.button:disabled {

  cursor: not-allowed;

  /* override the hover/focus values */
  --primary-color: var(--eds-color-neutral-300);
  --plain-background-color: transparent;
}
