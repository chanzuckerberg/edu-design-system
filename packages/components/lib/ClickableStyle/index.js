"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = void 0;
var ClickableStyle_1 = require("./ClickableStyle");
Object.defineProperty(exports, "default", { enumerable: true, get: function () { return __importDefault(ClickableStyle_1).default; } });
//# sourceMappingURL=index.js.map