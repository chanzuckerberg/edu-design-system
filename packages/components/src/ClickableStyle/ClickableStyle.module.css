.button {
  /* Base Styles */

  /* Using motion-reduce:transition-none here because motion-safe:transition-colors doesn't seem to
     be having an effect. :( */
  @apply rounded border-2 transition-colors motion-reduce:transition-none inline-flex items-center;

  /* Base Text Styles */
  @apply font-bold;
}

/* Component Sizes */
.sizeSmall {
  @apply px-4 py-2 text-xs leading-3;
}

.sizeMedium {
  @apply px-4 py-2 text-sm;
}

.sizeLarge {
  @apply px-6 py-4 text-sm;
}

/* Button Colors */
.colorBrand {
  --primary-color: var(--eds-color-brand-600);
  --secondary-color: var(--eds-color-white);
  --link-color: var(--eds-color-brand-700);

  &:hover,
  &.stateHover,
  &:focus,
  &.stateFocus {
    --primary-color: var(--eds-color-brand-700);
    --link-color: var(--eds-color-brand-800);
  }

  &:active,
  &.stateActive {
    --primary-color: var(--eds-color-brand-800);
  }

  /* override the hover/focus values */
  &:disabled {
    --primary-color: var(--eds-color-brand-300);
  }
}

.colorAlert {
  --primary-color: var(--eds-color-alert-600);
  --secondary-color: var(--eds-color-white);
  --link-color: var(--eds-color-alert-700);

  &:hover,
  &.stateHover,
  &:focus,
  &.stateFocus {
    --primary-color: var(--eds-color-alert-700);
    --link-color: var(--eds-color-alert-800);
  }

  &:active,
  &.stateActive {
    --primary-color: var(--eds-color-alert-800);
  }

  /* override the hover/focus values */
  &:disabled {
    --primary-color: var(--eds-color-alert-300);
  }
}

.colorNeutral {
  --primary-color: var(--eds-color-neutral-500);
  --secondary-color: var(--eds-color-white);
  --link-color: var(--eds-color-neutral-600);

  &:hover,
  &.stateHover,
  &:focus,
  &.stateFocus {
    --primary-color: var(--eds-color-neutral-600);
    --link-color: var(--eds-color-neutral-700);
  }

  &:active,
  &.stateActive {
    --primary-color: var(--eds-color-neutral-700);
  }

  /* override the hover/focus values */
  &:disabled {
    --primary-color: var(--eds-color-neutral-300);
  }
}

.colorSuccess {
  --primary-color: var(--eds-color-success-600);
  --secondary-color: var(--eds-color-white);
  --link-color: var(--eds-color-success-700);

  &:hover,
  &.stateHover,
  &:focus,
  &.stateFocus {
    --primary-color: var(--eds-color-success-700);
    --link-color: var(--eds-color-success-800);
  }

  &:active,
  &.stateActive {
    --primary-color: var(--eds-color-success-800);
  }

  /* override the hover/focus values */
  &:disabled {
    --primary-color: var(--eds-color-success-300);
  }
}

.colorWarning {
  --primary-color: var(--eds-color-warning-600);
  --secondary-color: var(--eds-color-white);
  --link-color: var(--eds-color-warning-700);

  &:hover,
  &.stateHover,
  &:focus,
  &.stateFocus {
    --primary-color: var(--eds-color-warning-700);
    --link-color: var(--eds-color-warning-800);
  }

  &:active,
  &.stateActive {
    --primary-color: var(--eds-color-warning-800);
  }

  /* override the hover/focus values */
  &:disabled {
    --primary-color: var(--eds-color-warning-300);
  }
}

.variantFlat {
  background-color: var(--primary-color);
  border-color: var(--primary-color);
  color: var(--secondary-color);
}

.variantOutline {
  background-color: transparent;
  border-color: var(--primary-color);
  color: var(--primary-color);

  &:focus,
  &.stateFocus {
    background-color: transparent;
    color: var(--primary-color);
  }

  &:hover,
  &.stateHover {
    background-color: var(--primary-color);
    color: var(--secondary-color);
  }

  &:disabled {
    background-color: transparent;
    color: var(--primary-color);
  }
}

.variantLink {
  @apply px-0 py-0 underline border-none font-bold;

  color: var(--link-color);

  /* button should defer to surrounding text for size/line-height/case */
  font-size: inherit;
  line-height: inherit;
  text-transform: inherit;

  &:hover,
  &.stateHover,
  &:focus,
  &.stateFocus {
    text-decoration-thickness: 2px;
  }

  &:active,
  &.stateActive,
  &:disabled {
    color: var(--primary-color);
  }

  /* override the hover/focus values */
  &:disabled {
    text-decoration-thickness: auto;
  }
}

/* comes after other styles to override hover behaviors */
.button:focus,
.button.stateFocus {
  /* Focus Outline */
  @apply outline-none ring-info-400 ring;
}

.button:disabled {
  @apply cursor-not-allowed;
}
