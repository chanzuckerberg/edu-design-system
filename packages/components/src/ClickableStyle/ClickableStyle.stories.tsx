import { Story } from "@storybook/react/types-6-0";
import React from "react";
import ClickableStyle from "./ClickableStyle";

export default {
  title: "ClickableStyle",
  component: ClickableStyle,
};

type Args = React.ComponentProps<typeof ClickableStyle>;

const Template: Story<Args> = (args) => <ClickableStyle {...args} />;

export const ClickableStyleAsALabelTag = Template.bind(null);
ClickableStyleAsALabelTag.args = {
  as: "label",
  color: "brand",
  variant: "flat",
  size: "large",
  children: "Label",
};
