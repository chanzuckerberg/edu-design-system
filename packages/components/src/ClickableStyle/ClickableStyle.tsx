import clsx from "clsx";
import React, { ReactNode } from "react";
import styles from "./ClickableStyle.module.css";

export type ClickableStyleProps<IComponent extends React.ElementType> = {
  /**
   * CSS class for custom styles.
   */
  className: string;
  /**
   * The color of the element.
   */
  color: "alert" | "brand" | "neutral" | "success" | "warning";
  /**
   * The size of the element.
   */
  size: "small" | "medium" | "large";
  /**
   * A hidden prop for visual testing
   */
  state?: "inactive" | "hover" | "focus" | "active";
  /**
   * The style of the element.
   */
  variant: "flat" | "outline" | "link";
  /**
   * Optional icon that can be positioned to the left or right of the other button contents.
   *
   * Default position is 'left'. Use the iconPlacement prop to move it to the right side.
   */
  icon?: {
    /*
     * The icon component. See the full list of available icons here:
     * https://chanzuckerberg.github.io/edu-design-system/?path=/story/svgicon--all-icons
     */
    iconElement: ReactNode;
    /*
     * Whether the icon should be on the left or right side of the button's other contents.
     *
     * Default is "left".
     *
     * If the button has no other contents, this prop will be ignored.
     */
    iconPlacement?: "left" | "right";
    /*
     * Margin to be added to icon's left and right sides.
     *
     * Icons typically have a little extra space between the path and the edge of the viewbox.
     * Negative left and right margin can be used to remove this extra space.
     */
    iconHorizontalMargin?: any;
  };
} & React.ComponentProps<IComponent>;

/**
 * A helper component that contains all the styling for buttons and links.
 *
 * See the Button stories for usage examples.
 */
const ClickableStyle = React.forwardRef(
  <IComponent extends React.ElementType>(
    {
      as: Component,
      children,
      color,
      size,
      state,
      variant,
      className,
      icon,
      ...rest
    }: ClickableStyleProps<IComponent>,
    ref: React.ForwardedRef<HTMLElement>,
  ) => {
    const maybeRenderIcon = (side: "left" | "right") => {
      if (!icon) {
        return;
      }

      const {
        iconElement,
        iconPlacement = "left",
        iconHorizontalMargin = 0,
      } = icon;

      if (
        (side === "left" && iconPlacement !== "left") ||
        (side === "right" && iconPlacement !== "right")
      ) {
        return null;
      }

      return (
        <div
          className={clsx(
            styles.icon,
            !!children && side === "left" && styles.iconLeft,
            !!children && side === "right" && styles.iconRight,
          )}
          style={{ margin: `0 ${iconHorizontalMargin}` }}
        >
          {iconElement}
        </div>
      );
    };

    return (
      <Component
        className={clsx(
          className,
          styles.button,
          // Sizes
          variant !== "link" && [
            size === "small" && styles.sizeSmall,
            size === "medium" && styles.sizeMedium,
            size === "large" && styles.sizeLarge,
          ],
          // Variants
          variant === "flat" && styles.variantFlat,
          variant === "outline" && styles.variantOutline,
          variant === "link" && styles.variantLink,
          // Colors
          color === "alert" && styles.colorAlert,
          color === "brand" && styles.colorBrand,
          color === "neutral" && styles.colorNeutral,
          color === "success" && styles.colorSuccess,
          color === "warning" && styles.colorWarning,
          // Interactive States (for testing)
          state === "hover" && styles.stateHover,
          state === "focus" && styles.stateFocus,
          state === "active" && styles.stateActive,
        )}
        ref={ref}
        {...rest}
      >
        {/* No width space to ensure height of contents */}
        {"\u200B"}

        {maybeRenderIcon("left")}

        {children}

        {maybeRenderIcon("right")}
      </Component>
    );
  },
);

ClickableStyle.displayName = "ClickableStyle";

export default ClickableStyle;
