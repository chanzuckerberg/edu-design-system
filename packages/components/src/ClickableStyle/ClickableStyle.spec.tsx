import { generateSnapshots } from "@chanzuckerberg/story-utils";
import * as ClickableStyleStoryFile from "./ClickableStyle.stories";

describe("<ClickableStyle />", () => {
  generateSnapshots(ClickableStyleStoryFile);
});
