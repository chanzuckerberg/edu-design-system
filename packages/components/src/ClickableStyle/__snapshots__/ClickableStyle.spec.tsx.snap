// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClickableStyle /> ClickableStyleAsALabelTag story renders snapshot 1`] = `
<label
  class="button sizeLarge variantFlat colorBrand"
>
  â€‹
  Label
</label>
`;
